sys/malloc_thread_safe/malloc_wrappers.c

```diff
@@ -29,18 +30,28 @@ static mutex_t _lock;
 void __attribute__((used)) *__wrap_malloc(size_t size)
 {
     assert(!irq_is_in());
-    mutex_lock(&_lock);
+    if (thread_get_active() != NULL) {
```

Comment:
I think due to the implementation of `thread_get_active()` the compiler will not be able to optimize out the second call to it. Maybe instead store the result in a variable?

That would also make it more obvious that the `mutex_unlock()` below is indeed only called when the corresponding `mutex_lock()` was called. (Even though I am pretty sure that the scheduler will not get enabled during the call to `__real_malloc()` e.g. from within an ISR now or in any future version of RIOT.)